package/base-files/files/lib/functions/caldata.sh

```diff
@@ -64,25 +76,30 @@ caldata_from_file() {
 
 	[ -n "$target" ] || target=/lib/firmware/$FIRMWARE
 
-	# dd doesn't handle partial reads from special files: use cat
-	cat $source | dd of=$target iflag=skip_bytes bs=$count skip=$offset count=1 2>/dev/null || \
+	caldata_dd $source $offset $count $target || \
 		caldata_die "failed to extract calibration data from $source"
 }
 
 caldata_sysfsload_from_file() {
 	local source=$1
 	local offset=$(($2))
 	local count=$(($3))
+	local target="/sys/$DEVPATH/data"
+	local target_dir=$(dirname "$target")
+	local ret
 
-	# dd doesn't handle partial reads from special files: use cat
-	# test extract to /dev/null first
-	cat $source | dd of=/dev/null iflag=skip_bytes bs=$count skip=$offset count=1 2>/dev/null || \
-		caldata_die "failed to extract calibration data from $source"
+	[ -d "$target_dir" ] || \
+		caldata_die "no sysfs dir to write: $target"
 
-	# can't fail now
 	echo 1 > /sys/$DEVPATH/loading
-	cat $source | dd of=/sys/$DEVPATH/data iflag=skip_bytes bs=$count skip=$offset count=1 2>/dev/null
-	echo 0 > /sys/$DEVPATH/loading
+	caldata_dd $source $offset $count $target
+	ret=$?
+	if [ ! $ret -eq 0 ]; then
```

Comment:
Minor nitpick - any reason to not simply do `[ $? != 0 ]` or even `if ! caldata_dd ...; then` ? We're not reusing `$ret` later and the syntax above is hard to understand (does it mean `(! $ret) == 0` or `!($ret == 0)`)

Reply:
Changed to `[  $? != 0 ]`, which make far more sense. Thank you!